miniapps/autodiff/taddensemat.hpp

```diff
@@ -0,0 +1,513 @@
+// Copyright (c) 2010-2021, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#ifndef TADDENSEMATRIX_H
+#define TADDENSEMATRIX_H
+
+#include "mfem.hpp"
+#include "tadvector.hpp"
+
+namespace mfem
+{
+/// Templated dense matrix data type.
+/** The main goal of the TADDenseMatrix class is to serve as a data
```

Comment:
What is `TAD` an abbreviation for?

Reply:
Template class aiding AD. Now it is changed to TAutoDiffDenseMatrix.